app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -388,5 +405,41 @@ void handleDataDelivery(RightMeshEvent e) {
             unDeliveredMessageIds.remove(deliveryDataId);
             updateInterface();
         }
+        else if(undeliveredPeerUpdateMessages.containsKey(deliveryDataId)){
+            // removing the message from undeliveredPeerUpdate map since we dont need to send it again.
+            undeliveredPeerUpdateMessages.remove(deliveryDataId);
+        }
+    }
+
+    /**
+     * Runs a timer task when there are undelivered peer updates in the que
+     */
+    void resendUndeliveredPackages(){
+        if(!isTimerRunning && !undeliveredPeerUpdateMessages.isEmpty()){
+            isTimerRunning = true;
+            undeliveredPackageTimer.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    //check if there are any undelivered packages
+                    if(!undeliveredPeerUpdateMessages.isEmpty()){
+                        byte[] message = createPeerUpdatePayloadFromUser(user);
+                        for(Map.Entry<Integer,MeshId> meshIdEntry: undeliveredPeerUpdateMessages.entrySet()){
+                            MeshId peerId = meshIdEntry.getValue();
+                            try {
+                                //send the packages again if there are undelivered packages
+                                meshManager.sendDataReliable(peerId, MESH_PORT, message);
```

Comment:
Does this duplicate the code where we first send these packets? Can they be shared between a function?

Reply:
do you mean put sendDataReliable inside a function so not to repeat it?

Reply:
No, like the creating a peerupdate message and looping over all the peers and sending it to them. The loop basically - is that all duplicate code?

Reply:
I see I dont think we are duplicating anything since since we are creating the byte message outside the loop and reusing it. inside forloop we get a new meshId each time and send the message to the id.

Reply:
Take a look at the `broadcastProfile()` method, that's the code I'm thinking is duplicated.